From 587f69ff37b4c9b6ba38c4778318ac18892d6ec5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 13 Apr 2022 08:30:48 +0300
Subject: [PATCH] [SW-0] common: move fw version offset device specific

In gaudi3, firmware version fuse offset is shifted by once row.
This patch moves firmware version offset to device specific files.

Change-Id: Icccf092af6623e6b248404cc575987fa1d81f48d
---

diff --git a/arch/gaudi2/arc/dev_efuse.h b/arch/gaudi2/arc/dev_efuse.h
index c92d21a..b7eac3f 100644
--- a/arch/gaudi2/arc/dev_efuse.h
+++ b/arch/gaudi2/arc/dev_efuse.h
@@ -4,6 +4,7 @@
  */
 
 #define EFUSE_KEY_HASH_OFFSET				0x14
+#define EFUSE_SECURITY_FW_VERSION_OFFSET		0x3B
 
 /*
  * 00: ROM, 01: EROM, 10: RSVD, 11:ROM
diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index 71a9f99..5d7e4ca 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -9,6 +9,7 @@
 /* Unused for gaudi3. Maintained for common code compilation */
 #define EFUSE_KEY_HASH_OFFSET				0x0
 #define EFUSE_SYSCFG2_OFFSET				0x3B
+#define EFUSE_SECURITY_FW_VERSION_OFFSET		0x3C
 
 #define EFUSE_SYSCFG_REVOKE_KEY_0			BIT(0)
 #define EFUSE_SYSCFG_PERMIT_KEY_4			~BIT(4)
diff --git a/arch/greco/arc_elf32/dev_efuse.h b/arch/greco/arc_elf32/dev_efuse.h
index c2963a0..bd864a9 100644
--- a/arch/greco/arc_elf32/dev_efuse.h
+++ b/arch/greco/arc_elf32/dev_efuse.h
@@ -4,6 +4,7 @@
  */
 
 #define EFUSE_KEY_HASH_OFFSET			0x12
+#define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 #define EFUSE_SYSCFG_ENABLE_4B_TERMINATION	BIT(25)
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
diff --git a/common/efuse.h b/common/efuse.h
index 47316a3..cd4567b 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -15,7 +15,6 @@
 #define EFUSE_NOT_READ				0x1
 
 #define EFUSE_SYSCFG_OFFSET			0x3A
-#define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 
 #define EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL	BIT(24)
 #define EFUSE_SYSCFG_WA_SNPS_CASE_01038891	BIT(23)
